src/main/scala/firrtl/Emitter.scala

```diff
@@ -515,7 +523,48 @@ class VerilogEmitter extends SeqTransform with Emitter {
     def declareVectorType(b: String, n: String, tpe: Type, size: BigInt, info: Info) = {
       declares += Seq(b, " ", tpe, " ", n, " [0:", bigIntToVLit(size - 1), "];", info)
     }
-
+    def declareVectorType(b: String, n: String, tpe: Type, size: BigInt, info: Info, preset: Expression) = {
+      declares += Seq(b, " ", tpe, " ", n, " [0:", bigIntToVLit(size - 1), "] = ", preset, ";", info)
+    }
+    
+    val moduleTarget = CircuitTarget(circuitName).module(m.name)
+    
+    /** 
+      * Store Emission option per Target
+      * Guarantee only one emission option per Target
+      * Named is used as base Class for all Target supported in SingleTargetAnnotation  
+      */
+    case class EmissionOptionMap[V <: EmissionOption](val df : V) extends collection.mutable.HashMap[Named, V] {
+      override def default(key: Named) = df
+      override def +=(elem : (Named, V)) : EmissionOptionMap.this.type = {
+        if (this.contains(elem._1))
+          throw EmitterException(s"Multiple EmissionOption for the target ${elem._1} (${this(elem._1)} ; ${elem._2})")
+        super.+=(elem)
+      } 
+    }
+    
+    val registerEmissionOption = EmissionOptionMap[RegisterEmissionOption](RegisterEmissionOptionDefault)
+    val wireEmissionOption = EmissionOptionMap[WireEmissionOption](WireEmissionOptionDefault)
+    val portEmissionOption = EmissionOptionMap[PortEmissionOption](PortEmissionOptionDefault)
+    val nodeEmissionOption = EmissionOptionMap[NodeEmissionOption](NodeEmissionOptionDefault)
+    val connectEmissionOption = EmissionOptionMap[ConnectEmissionOption](ConnectEmissionOptionDefault)
+    val emissionAnnos = annotations.collect{ case m : SingleTargetAnnotation[Named] with EmissionOption => m }
+    // using multiple foreach instead of a single partial function as an Annotation can gather multiple EmissionOptions for simplicity
+    emissionAnnos.foreach { case a :RegisterEmissionOption => registerEmissionOption += ((a.target,a)) case _ => }
+    emissionAnnos.foreach { case a :WireEmissionOption     => wireEmissionOption += ((a.target,a))     case _ => }
+    emissionAnnos.foreach { case a :PortEmissionOption     => portEmissionOption += ((a.target,a))     case _ => }
+    emissionAnnos.foreach { case a :NodeEmissionOption     => nodeEmissionOption += ((a.target,a))     case _ => }
+    emissionAnnos.foreach { case a :ConnectEmissionOption  => connectEmissionOption += ((a.target,a))  case _ => }
```

Comment:
This logic occurs for each module emitted (A `VerilogEmitter` is created for each `Module`). I think I'd rather that this logic be embedded inside a class that is then passed as an argument to the `VerilogRenderer` so we only do it once. Something like:
```scala
class EmissionOptions(annos: AnnotationSeq) {
  // Private so that we can present an immutable API
  private val registerEmissionOption = EmissionOptionMap[RegisterEmissionOption](RegisterEmissionOptionDefault)
  ...

  def getRegisterEmissionOption(target: ReferenceTarget): RegisterEmissionOption =
    registerEmissionOption(target)

  // We could tweak this logic to make this go away after initial creation but it's not a huge deal
  private val emissionAnnos = annotations.collect{ case m : SingleTargetAnnotation[Named] with EmissionOption => m }
  // using multiple foreach instead of a single partial function as an Annotation can gather multiple EmissionOptions for simplicity
  emissionAnnos.foreach { case a :RegisterEmissionOption => registerEmissionOption += ((a.target,a)) case _ => }
  ...
}
```

Reply:
done see https://github.com/freechipsproject/firrtl/pull/1050/commits/b47cae0340024d26ca9673dd79561d007f2ecaef